client/service_create_test.go

```diff
@@ -121,3 +122,94 @@ func TestServiceCreateCompatiblePlatforms(t *testing.T) {
 		t.Fatalf("expected `service_amd64`, got %s", r.ID)
 	}
 }
+
+type pinByDigestTestParams struct {
+	img      string // input image provided by the user
+	expected string // expected image after digest pinning
+}
+
+var dgst = "sha256:c0537ff6a5218ef531ece93d4984efc99bbf3f7497c0a7726c88e2bb7584dc96"
+var pinByDigestTests = []pinByDigestTestParams{
+	// default registry returns familiar string
+	{"docker.io/library/alpine", "alpine:latest@" + dgst},
+	// provided tag is preserved and digest added
+	{"alpine:edge", "alpine:edge@" + dgst},
+	// image with provided digest remains unchanged
+	{"alpine@" + dgst, "alpine@" + dgst},
+	// image with provided tag and digest remains unchanged
+	{"alpine:edge@" + dgst, "alpine:edge@" + dgst},
```

Comment:
How about providing a different digest from `dgst` here and making sure it isn't changed to `dgst`?